apps/utils/models.py

```diff
@@ -1,7 +1,25 @@
 from django.db import models
 
 
-class BaseModel(models.Model):
+class VersioningMixin:
+    def compare_with_model(self, new: models.Model, exclude_fields: list[str]) -> set:
```

Comment:
how does this handle foreign keys and many-to-many relationships?

Reply:
Good question.
FK: Since this is just a number, the two FK id's are compared.
M2M: This is currently not being handled, since it's not being returned by the `instance._meta.fields` iterator. I'll need to update the code to handle this as well, especially since we now do compare two experiments when we compare routes. I'm thinking of maybe just comparing IDs, which should be enough.